### ISSUE TITLE
remote_file doesn't work

### ISSUE THREAD

**Issue by rowillia** [facebook/buck#408]
Using the example code from https://buckbuild.com/rule/remote_file.html

```
remote_file(
  name = 'jetty-source',
  out = 'jetty-all-9.2.10.v20150310-sources.jar',
  url = 'mvn:org.eclipse.jetty.aggregate:jetty-all:src:9.2.10.v20150310',
  sha1 = '311da310416d2feb3de227081d7c3f48742d7075',
)
```

It appears the `optionalServer` isn't optional.

**Comment by sdwilsh** [comment]
Have you set `download.maven_repo` (which appears to be undocumented) in your `.buckconfig`?  I believe it doesn't require it to be set as long as it is set there.

**Comment by rowillia** [comment]
I did, the problem is src/com/facebook/buck/file/MavenUrlDecoder.java expects there to always be 6 fields.

**Comment by sdwilsh** [comment]
Sounds like we were missing a test case then :/

**Comment by rowillia** [comment]
https://github.com/facebook/buck/pull/411 fixes this.

**Comment by sdwilsh** [comment]
Closed by #411.
